pkg/cmd/monitor/monitor.go

```diff
@@ -0,0 +1,279 @@
+package monitor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/cmd/monitor/health"
+	"github.com/openshift/library-go/pkg/serviceability"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/pkg/transport"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+const (
+	DefaultHTTPDialTimeout     = 2 * time.Second
+	DefaultHealthCheckInterval = 5 * time.Second
+	DefaultEndpoint            = "https://localhost:2379"
+	DefaultLogLevel            = 2
+	DefaultLogOutputs          = "stderr"
+	DefaultStaticPodVersion    = 0
+
+	// DefaultLogRotationConfig is the default configuration used for log rotation.
+	// Log rotation is disabled by default.
+	// MaxSize    = 100 // MB
+	// MaxAge     = 0 // days (no limit)
+	// MaxBackups = 10
+	// LocalTime  = false // use computers local time, UTC by default
+	// Compress   = true // compress the rotated log in gzip format
+	DefaultLogRotationConfig = `{"maxsize": 100, "maxage": 0, "maxbackups": 10, "localtime": false, "compress": true}`
+)
+
+// defaults used by kube-apiserver
```

Comment:
if this starts working nicely, we may want to improve granularity by choosing a smaller value.